ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/Evaluator.java

```diff
@@ -58,17 +60,30 @@ protected Evaluator(Preprocessor preprocessor, TreeParser treeParser,
     public abstract void initialize() throws BallerinaShellException;
 
     /**
-     * Base evaluation function which evaluates an input line.
+     * Base evaluation function which returns compilation.
      * <p>
      * An input line may contain one or more statements separated by semicolons.
      * The result will be written via the {@code ShellResultController}.
      * <p>
      * If the execution failed, an error will be thrown instead.
      *
      * @param source Input line from user.
+     * @return compilation.
+     */
+    public abstract ShellCompilation getCompilation(String source);
+
+    /**
+     * Base evaluation function.
+     * <p>
+     * An input line may contain one or more statements separated by semicolons.
+     * The result will be written via the {@code ShellResultController}.
+     * <p>
+     * If the execution failed, an error will be thrown instead.
+     *
+     * @param compilation compilation.
      * @return String output from the evaluator. This will be the last output.
      */
-    public abstract String evaluate(String source) throws BallerinaShellException;
+    public abstract String getValue(Optional<PackageCompilation> compilation) throws BallerinaShellException;
```

Comment:
Since this is a public API, without changing the signature, let's add a newAPI for `getValue` keeping `evaluate` API as well

Reply:
Updated in 9f270db8b7259fb63e77545e8454623db4d6c3a7